### PR TITLE
docs: correct default application builder for new apps

### PR DESCRIPTION
**Pull request by xfh**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The documentation on angular.dev mentions `@angular-devkit/build-angular:browser` as default application builder. This is not correct. Recent applications are setup with build-angular:application by default.

## What is the new behavior?

The docs name `@angular-devkit/build-angular:application` as the default application builder.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: adev/src/content/tools/cli/build.md [hunk 1]
```diff
@@ -41,6 +41,9 @@ You can determine which builder is being used for a particular project by lookin
 
 This page discusses usage and options of `@angular-devkit/build-angular:browser`.
 
+The properties for the other builders are mostly the same. 
+For differences, see [`browser-esbuild` documentation](tools/cli/esbuild).
+
```
Comment (alan-agius4) [inline adev/src/content/tools/cli/build.md:46]:
```suggestion
This page discusses usage and options of `@angular-devkit/build-angular:application`.

```

Comment (xfh) [reply adev/src/content/tools/cli/build.md]: I've added a fixup with the suggested changes.

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks

**Comment by alan-agius4** [comment]
**caretaker note:** my approval should be sufficient for this change.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit d181a36296a6f2b12bb2977c602b2738513fc2b5.

**Pull request closed without merging**